tests/explainers/test_tree.py

```diff
@@ -1122,3 +1000,164 @@ def test_xgboost_buffer_strip():
     # after this fix, this line should not error
     explainer = shap.TreeExplainer(model)
     assert isinstance(explainer, shap.explainers.Tree)
+
+
+class TestExplainerLightGBM:
```

Comment:
Is grouping the tests into a single class just for organization? Perhaps this should just be its own module? I don't think the TestClass paradigm is especially common in the codebase? I'd be more open to using a class if we were [reusing fixtures and/or parametrizations](https://stackoverflow.com/questions/50016862/grouping-tests-in-pytest-classes-vs-plain-functions). If we are going to organize into a single class, should we pull out common fixtures like `shap.datasets.california`?

Reply:
Yeah, it's just for organization. Because at the moment, there's too many tests in `test_tree.py` for all the various supported tree models.

I'm open to changing it to a separate file rather than organizing as a class. The benefit of _not_ doing so is just that we get to maintain a 1-1 mapping between the test file and the code file, e.g. `test_tree.py` and `_tree.py`. It makes the relevant tests easier to find.  
And if the code file (like `_tree.py`) supports multiple scenarios / independent cases (like xgboost,catboost, ...), then we can organize within the test file using classes.

It's not a common paradigm in the codebase because this (^) really only happens for the `_tree` module. Perhaps that means we should refactor `_tree.py` instead, but that's an even bigger change so maybe a less suitable option now.

> If we are going to organize into a single class, should we pull out common fixtures like shap.datasets.california?

Yes we could, if it is sufficiently common. I can do this if we decide to maintain the TestClass paradigm. Let me know.

Reply:
> we get to maintain a 1-1 mapping between the test file and the code file, e.g. test_tree.py and _tree.py

That's true.

> Perhaps that means we should refactor _tree.py instead, but that's an even bigger change, so maybe a less suitable option now.

Agreed. 

> Yes we could, if it is sufficiently common. I can do this if we decide to maintain the TestClass paradigm. Let me know.

Sounds good. Let's keep an eye out for the opportunity to share fixtures where possible. 

Approved!